packages/markdown-docx/src/ToOOXMLVisitor/index.js

```diff
@@ -51,6 +52,23 @@ class ToOOXMLVisitor {
         this.tags = [];
         // Relationship tags for links in a document
         this.relationships = [];
+        // Contains the tags for optional or conditional nodes.
+        // Optional/conditional themselves can act contain multiple inline nodes.
+        // However, they are not block elements and are present inside
+        // paragraph or clause nodes so an extra array is needed to store their info.
```

Comment:
Even a paragraph contains multiple nodes but we don't maintain a different array for it. You're using a DFS approach which should technically allow you to traverse each optional node _first_ and then it would move forward. I will try to write a pseudo code for it and I hope it gives you an idea.

```
traverseNodes() {
   if OPTIONAL { 
      ooxmlToBeDisplayed = traverseNodes(whenSome) // ofcouse this depends upon `hasSome` so you would need a check for it.
      ooxmlToBeHidden = travserseNodes(whenNode)
  }
}
```

I didn't bother with `relationships` because I knew you had to store them in a different part of OOXML (some node above `w:document`) so you needed to store them until your transformation is over.

Reply:
I think this would require a refactoring of `traverseNodes` as it doesn't return anything unlike `toCiceroMark`. Maybe you need to keep that on priority because the current code is getting out of hand in terms of readability.